48522 5 tidy up

ricellis opened the pull request:
*What*
Complete the Apache HttpClient replacement work

*How*
* Removed Apache HttpClient dependency from gradle
* Removed Apache commons-logging
* Added support for deserializing the error stream into `CouchDbException` subclasses, which in some cases adds additional useful info to the exception.
* Removed the code in `HttpConnection` that was disabling the JVM persistent http connections

*Testing*
* Minor test code cleanups
* Added new test for `Expect: 100-continue` behaviour
* Changed test JVM to `openjdk7`

reviewer @emlaver 
reviewer @alfinkel

alfinkel commented on src/main/java/com/cloudant/client/api/Database.java:
```diff
@@ -86,7 +83,7 @@
 public class Database {
 
-    static final Log log = LogFactory.getLog(Database.class);
+    static final Logger log = Logger.getLogger(Database.class.getName());
```
I think you might want to use `Database.class.getCanonicalName()` here instead of `Database.class.getName()` since it seems to be more appropriate for logging operations.  See:  http://stackoverflow.com/questions/15202997/what-is-the-difference-between-canonical-name-simple-name-and-class-name-in-jav.

ricellis replied:
I think using `getName()` would be the same in these cases anyway, but I have no objection to using `getCanonicalName()` and making things easier for any future additions. Fixed in 0bbde07.

alfinkel commented on src/main/java/com/cloudant/client/api/Search.java:
```diff
@@ -85,7 +83,7 @@
 public class Search {
 
-    private static final Log log = LogFactory.getLog(Search.class);
+    private static final Logger log = Logger.getLogger(Search.class.getName());
```
Same comment as https://github.com/cloudant/java-cloudant/pull/120/files#r43167135

alfinkel commented on src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java:
```diff
@@ -43,7 +41,7 @@
 public abstract class CouchDatabaseBase {
 
-    static final Log log = LogFactory.getLog(CouchDatabase.class);
+    static final Logger log = Logger.getLogger(CouchDatabase.class.getName());
```
Same comment as https://github.com/cloudant/java-cloudant/pull/120/files#r43167135

alfinkel commented on src/main/java/com/cloudant/client/org/lightcouch/CouchDbClient.java:
```diff
@@ -60,7 +58,7 @@
 public class CouchDbClient {
 
-    static final Log log = LogFactory.getLog(CouchDbClient.class);
+    static final Logger log = Logger.getLogger(CouchDbClient.class.getName());
```
Same comment as https://github.com/cloudant/java-cloudant/pull/120/files#r43167135

alfinkel commented on src/main/java/com/cloudant/client/org/lightcouch/CouchDbConfig.java:
```diff
@@ -28,7 +26,7 @@
  */
 class CouchDbConfig {
-    private static final Log log = LogFactory.getLog(CouchDbConfig.class);
+    private static final Logger log = Logger.getLogger(CouchDbConfig.class.getName());
```
Same comment as https://github.com/cloudant/java-cloudant/pull/120/files#r43167135

alfinkel commented on src/main/java/com/cloudant/client/org/lightcouch/Replication.java:
```diff
@@ -58,7 +57,7 @@
 public class Replication {
 
-    static final Log log = LogFactory.getLog(Replication.class);
+    static final Logger log = Logger.getLogger(Replication.class.getName());
```
Same comment as https://github.com/cloudant/java-cloudant/pull/120/files#r43167135

emlaver commented:
@ricellis There is still a TODO comment in `CloudantClientTests` for a client initialized with the cookie. I believe we can remove the `cookieBasedClient` variable and all its references since there's no difference between it and `account`:
https://github.com/cloudant/java-cloudant/blob/feature-http-replacement/src/test/java/com/cloudant/tests/CloudantClientTests.java#L62-L70

ricellis commented:
@emlaver I fixed the TODO in 24f5313.
I also noticed that because of the interceptors and cookie changes there was a bunch of un-needed code left hanging around for `CouchDbClient` configuration, which I have removed in 6caa93e98f6761fdff610f08802e6fa60b6167f3

alfinkel commented:
+1

emlaver commented:
Looks good - thanks for catching the other fixes.  +1
